experimental/AlgebraicStatistics/src/PhylogeneticAuxiliary.jl

```diff
@@ -0,0 +1,71 @@
+##########################
+#### GROUP OPERATIONS ####
+##########################
+
+function group_sum(pm::GroupBasedPhylogeneticModel, states::Vector{Int})
+  group = group_of_model(pm)
+  return sum(group[states])
+end
+
+function is_zero_group_sum(pm::GroupBasedPhylogeneticModel, states::Vector{Int})
+  return group_sum(pm, states) == zero(parent(group_of_model(pm)[1]))
+end
+  
+function which_group_element(pm::GroupBasedPhylogeneticModel, elem::FinGenAbGroupElem)
+  group = group_of_model(pm)
+  return findall([all(g==elem) for g in group])[1]
+end
+
+
+########################################
+#### AUXILIARY FUNCTIONS FOR GRAPHS ####
+########################################
+
+function interior_nodes(graph::Graph)
+  big_graph = Polymake.graph.Graph(ADJACENCY = pm_object(graph))
+  degrees = big_graph.NODE_DEGREES
+  return findall(x -> x > 1, degrees)
+end
+
+function leaves(graph::Graph)
+  big_graph = Polymake.graph.Graph(ADJACENCY = pm_object(graph))
+  degrees = big_graph.NODE_DEGREES
+  return findall(x -> x == 1, degrees)
+end
+
+function vertex_descendants(v::Int, gr::Graph, desc::Vector{Any})
+  lvs = leaves(gr)
+  outn = outneighbors(gr, v)
+
+  if v in lvs
+    return [v]
+  end
+
+  innodes = setdiff(outn, lvs)
+  d = unique(append!(desc, intersect(outn, lvs)))
+  
+  if length(innodes) > 0
+    for i in innodes
+      d = vertex_descendants(i, gr, d)
+    end
+    return d 
+  end
+
+  return d
+end
+
+function root(graph::Graph)
+  n_parents = [length(inneighbors(graph, v)) for v in 1:n_vertices(graph)]
+  return findall(x -> x == 0, n_parents)[1]
+end
```

Comment:
Is it important whether these are directed or undirected? Because when you convert you always get something undirected:
```
julia> g = Graph{Directed}(5)
Directed graph with 5 nodes and no edges

julia> add_edge!(g,1,2)
true

julia> pmg = Polymake.graph.Graph(ADJACENCY=Oscar.pm_object(g))
type: Graph<Undirected>

ADJACENCY
	{1}
	{0}
	{}
	{}
	{}
```
You could instead forward the template argument:
```
julia> pmg = Polymake.graph.Graph{Directed}(ADJACENCY=Oscar.pm_object(g))
type: Graph<Directed>

ADJACENCY
	{1}
	{}
	{}
	{}
	{}
```